### DISCUSSION TITLE
Anyone developing an Android TV app for Stash?

### DISCUSSION THREAD

**Discussion by Waldorf3** [stashapp/stash#3844] [Ideas]
Yeah yeah I know, Google will never allow it on Google Play, but that should not stop it from being emmensely popular - just look at https://github.com/yuliskov/SmartTubeNext.

**Comment by andrewmeyer** [comment]
+1 to this, it can be compiled and distributed as a build asset

**Comment by DogmaDragon** [comment]
Nobody is working one as far as I know. If the TV supports apps, then it has browser functionality, which is all you need to open Stash on any device. 

A dedicated app would only really improve the navigation, but might even limit the range of supported codecs. So it's not even a clear improvement overall. And that adds extra overhead for each new release, as it's an additional platform that needs to be ported for.

**Comment by scruffynerf** [comment]
Any app can add Stash access, it's a graphql API...
So hopefully some will write one but it won't be part of the main app, it's gonna be a UI client only.

**Comment by mat926** [comment]
> Yeah yeah I know, Google will never allow it on Google Play, but that should not stop it from being emmensely popular 

Who says that they have to market it as a porn app?

**Comment by dingdongsolong4** [comment]
> Who says that they have to market it as a porn app?

Well that is kinda what it is... but either way, that isn't really the reason why no app has been developed. As @DogmaDragon mentioned, it's an additional platform to support and maintain, and we don't have infinite resources here. For an Android TV app to make sense, it would need to be geared towards use on a TV. In other words, a lot of the UI would have to be rebuilt from scratch, which is not an easy task.

But following on from @scruffynerf's comment, the Stash UI is just a frontend for the GraphQL API exposed by the server. There is nothing preventing anyone from writing their own Stash UI - whether that be an alternative web-based UI, or an iOS app, or an Android TV app. It would be really great if someone made an Android TV app, but such an app is unlikely to be officially developed as part of Stash anytime soon.

**Comment by scruffynerf** [comment]
In the meantime, Kodi is widely supported on Android and other platforms, and with https://github.com/scruffynerf/StashNfoExporterKodi you can easily make a library of your Stash content, and either use Stash to serve them, or have Kodi access the files directly, either works well.

**Comment by damontecres** [comment]
Hey all! I've actually started to create a basic Android TV app for Stash! Check it out: https://github.com/damontecres/StashAppAndroidTV

It is open-source under AGPLv3 like Stash itself. It is not published on any App store, so it must be side loaded (see instructions in the link above).

The app is pretty basic and a lot of features of Stash are not supported, but it works okay! I've tested with my Nvidia Shield TV (and I assume it will work on other devices too). Also, it's my first Android TV app, so the code is a bit messy and the UI/UX is not great, but I'm working on improvements!

Feedback and contributions are welcome!

**Reply by ChilledSlim** [reply]
This is unbelievable.  The app functions perfectly from my testing and is cleanly laid out and quite impressive. Will see what I can contribute to it.

**Reply by keenbed** [reply]
Also installed it and it works out of the box, great work!

**Reply by KaoriRinaFujimoto** [reply]
indeed, awesome job! The app works really, really good! Big thanks

**Comment by sashapp** [comment]
Hello there :)
I wrote an Android mobile device client for StashDB.

https://play.google.com/store/apps/details?id=ch.avocode.sash

![Screenshot 2025-05-06 at 16 11 57](https://github.com/user-attachments/assets/b05899ed-2efe-41d1-a4ef-55b5f744dcf0)

While Sash is a client for browsing scenes and performers on StashDB (and not Stash app), it offers an interface to your instance of Stash app. It also allows to connect to Prowlarr to bridge the gap between the scene listings and downloading them.

While browsing StashDB is free, I put Prowlarr and Stash app connections behind a (cheap) paywall, to support myself in the development that is to come. In the future, I plan to add casting support (such as Chromecast) to get videos on any screen, as well as a more comprehensive interface to Stash app.

Feedback is very much welcome. The app is new and I plan on keeping development going.
